vue-components/src/components/InputWithExtender.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+	<div
+		:class="[ 'wikit', 'wikit-InputWithExtender' ]"
+	>
+		<TextInput
+			:value="value"
+			@input="onInput"
+			:label="label"
+			@focus="showExtension = true"
+			@blur="showExtension = false"
+			:error="error"
+			:placeholder="placeholder"
+			:disabled="disabled"
+		/>
+		<div v-if="showExtension" class="wikit-InputWithExtender__extension">
+			<slot />
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import VueCompositionAPI, { defineComponent } from '@vue/composition-api';
+import TextInput from '@/components/TextInput.vue';
+import { errorProp } from '@/compositions/validatable';
+
+Vue.use( VueCompositionAPI );
+
+export default defineComponent( {
+	name: 'InputWithExtender',
+	inheritAttrs: false,
+	components: {
+		TextInput,
+	},
+	data() {
+		return {
+			showExtension: false,
+		};
+	},
+	props: {
+		error: errorProp,
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+		placeholder: {
+			type: String,
+			default: '',
+		},
+		value: {
+			type: String,
+			default: null,
+		},
+		label: {
+			type: String,
+			default: null,
+		},
+	},
+	methods: {
+		onInput( value: string ): void {
+			this.showExtension = true;
+			this.$emit( 'input', value );
+		},
+		onEsc(): void {
```

Comment:
This is unused, but it is functionality that we want.

Reply:
Looks good but doesn't work :upside_down_face: 

I think you may have to emit that event inside the TextInput as well

Reply:
just a heads-up that this seems to not be working any more